addons/payment_ingenico/models/payment.py

```diff
@@ -29,13 +30,27 @@ class PaymentAcquirerOgone(models.Model):
     ogone_pspid = fields.Char('PSPID', required_if_provider='ogone', groups='base.group_user')
     ogone_userid = fields.Char('API User ID', required_if_provider='ogone', groups='base.group_user')
     ogone_password = fields.Char('API User Password', required_if_provider='ogone', groups='base.group_user')
-    ogone_shakey_in = fields.Char('SHA Key IN', size=32, required_if_provider='ogone', groups='base.group_user')
-    ogone_shakey_out = fields.Char('SHA Key OUT', size=32, required_if_provider='ogone', groups='base.group_user')
+    ogone_shakey_in = fields.Char('SHA Key IN', required_if_provider='ogone', groups='base.group_user',
+                                  help="SHA key linked to your Ogone account should be of length 32"
+                                  " if the hash algorithm is SHA1, 64 for SHA256 or 128 for SHA512")
+    ogone_shakey_out = fields.Char('SHA Key OUT', required_if_provider='ogone', groups='base.group_user',
+                                   help="SHA key linked to your Ogone account should be of length 32"
+                                   " if the hash algorithm is SHA1, 64 for SHA256 or 128 for SHA512")
     ogone_alias_usage = fields.Char('Alias Usage', default="Allow saving my payment data",
                                     help="If you want to use Ogone Aliases, this default "
                                     "Alias Usage will be presented to the customer as the "
                                     "reason you want to keep his payment data")
 
+    # TODO : Add selection field in master
+    @api.onchange('ogone_shakey_in', 'ogone_shakey_out')
+    def _onchange_shakey(self):
+        if self.ogone_shakey_in and len(self.ogone_shakey_in) not in [32, 64, 128]:
+            raise UserError(_('%s should be of length 32, 64 or 128 caracters') % _('SHA Key IN'))
```

Comment:
```suggestion
            raise UserError(_('SHA Key IN of lenght %s, expect 32, 64 or 128 caracters') % len(self.ogone_shakey_in))
```
not sure why you split in two. Also, always a good idea to add information for debugging.

Reply:
The goal for splitting in two is to add only a generic line in the translation files.
+ Will change the UserError to what you suggest

Reply:
This was a good intention but it's actually a bad idea, it's making the translators live harder.
https://www.odoo.com/documentation/15.0/developer/misc/i18n/translations.html#blocks